inttests/volume_test.go

```diff
@@ -476,3 +620,87 @@ func TestSetMappedSdcLimitsInvalid(t *testing.T) {
 	assert.Nil(t, err)
 	deleteAllVolumes(t)
 }
+
+// Testing TestLockUnlockAutoSnapshot will attempting locking the auto snapshot and unlocking the auto snapshot
+func TestLockUnlockAutoSnapshot(t *testing.T) {
+	volID, err := createVolume(t, "")
+	assert.Nil(t, err)
+	newVolume, err := getVolByID(volID)
+	assert.Nil(t, err)
+
+	// now make a snapshot
+	snapshotDefs := make([]*siotypes.SnapshotDef, 0)
+	snapname := fmt.Sprintf("%s-%s", newVolume.Name, "snap")
+
+	snapDef := &siotypes.SnapshotDef{
+		VolumeID:     volID,
+		SnapshotName: snapname,
+	}
+	snapshotDefs = append(snapshotDefs, snapDef)
+	snapParam := &siotypes.SnapshotVolumesParam{
+		SnapshotDefs: snapshotDefs,
+	}
+
+	system := getSystem()
+	assert.NotNil(t, system)
+
+	// Create snapshot
+	snapResponse, err := system.CreateSnapshotConsistencyGroup(snapParam)
+	assert.Nil(t, err)
+	assert.NotZero(t, len(snapResponse.VolumeIDList))
+
+	// Get StoragePool
+	pool := getStoragePool(t)
+	volumes, err := pool.GetVolume("", volID, "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Get Snapshot
+	volumes, err = pool.GetVolume("", snapResponse.VolumeIDList[0], "", "", true)
+	assert.Nil(t, err)
+	assert.NotNil(t, volumes)
+
+	// Lock Snapshot
+	snap, err := getVolByID(volumes[0].ID)
+	assert.Nil(t, err)
+	sr := goscaleio.NewVolume(C)
+	sr.Volume = snap
+	err = sr.LockAutoSnapshot()
+	assert.NotNil(t, err)
+	err = sr.UnlockAutoSnapshot()
+	assert.NotNil(t, err)
+
+	// Delete Snapshot and Volume
+	err = deleteVolume(t, sr.Volume.ID)
+	assert.Nil(t, err)
+	err = deleteVolume(t, volID)
+	assert.Nil(t, err)
+	deleteAllVolumes(t)
+}
+
+// Testing TestSetVolumeAccessModeLimit will be attempting set access mode of volume
+func TestSetVolumeAccessModeLimit(t *testing.T) {
```

Comment:
The methods TestSetVolumeAccessModeLimit and TestSetVolumeAccessModeLimitInvalid are performing the same functionality just that one is mocking on valid value and other on invalid. Please optimize these calls into one method, as there is redundant code.
Refer to this test method for mocking on different scenarios: https://github.com/dell/csi-powerflex/blob/main/service/service_unit_test.go#L382

Reply:
Incorporated the review comment in this commit [3740c4e]
Thanks, Rajshree for your valuable feedback.